pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -360,6 +360,92 @@ func TestPriorityQueue_Delete(t *testing.T) {
 	}
 }
 
+func BenchmarkMoveAllToActiveOrBackoffQueue(b *testing.B) {
+	tests := []struct {
+		name                 string
+		podsInUnschedulableQ int
+		moveEvent            string
+	}{
+		{
+			name:      "baseline",
+			moveEvent: UnschedulableTimeout,
+		},
+		{
+			name:      "worst",
+			moveEvent: NodeAdd,
+		},
+		{
+			name: "random",
+			// leave "moveEvent" unspecified
+		},
+	}
+
+	podTemplates := []*v1.Pod{
+		highPriorityPodInfo.Pod, highPriNominatedPodInfo.Pod,
+		medPriorityPodInfo.Pod, unschedulablePodInfo.Pod,
+	}
+	events := []string{NodeAdd, NodeTaintChange, NodeAllocatableChange, NodeConditionChange, NodeLabelChange, PodAdd,
+		PvcAdd, PvcUpdate, PvAdd, PvUpdate, StorageClassAdd, CSINodeAdd, CSINodeUpdate}
+	pluginNum := 20
+	var plugins []string
+	// Mimic that we have 20 plugins loaded in runtime.
+	for i := 0; i < pluginNum; i++ {
+		plugins = append(plugins, fmt.Sprintf("fake-plugin-%v", i))
+	}
+
+	for _, tt := range tests {
+		for _, podsInUnschedulableQ := range []int{1000, 5000} {
+			b.Run(fmt.Sprintf("%v-%v", tt.name, podsInUnschedulableQ), func(b *testing.B) {
+				for i := 0; i < b.N; i++ {
+					b.StopTimer()
+					c := clock.NewFakeClock(time.Now())
+
+					m := make(map[framework.ClusterEvent]sets.String)
+					if tt.moveEvent != "" {
+						// Simulate that all plugins register the NodeAdd event.
+						m[clusterEventReg[NodeAdd]] = sets.NewString(plugins...)
+					} else {
+						// Random case.
+						for j := 0; j < len(events); j++ {
+							m[clusterEventReg[events[j]]] = sets.NewString()
+							for k := 0; k < len(plugins); k++ {
+								if (k+1)%(j+1) == 0 {
+									m[clusterEventReg[events[j]]].Insert(plugins[k])
+								}
+							}
+						}
+					}
+
+					q := NewPriorityQueue(newDefaultQueueSort(), WithClock(c), WithClusterEventMap(m))
+
+					// Init pods in unschedulableQ.
+					for i := 0; i < podsInUnschedulableQ; i++ {
+						p := podTemplates[i%len(podTemplates)].DeepCopy()
+						p.Name, p.UID = fmt.Sprintf("%v-%v", p.Name, i), types.UID(fmt.Sprintf("%v-%v", p.UID, i))
+						var podInfo *framework.QueuedPodInfo
+						if tt.moveEvent != "" {
+							// Each pod failed by all plugins.
```

Comment:
what's the rationale here?

Reply:
The (unschedulable) plugin names here will be eventually intersected with the plugins that register the moveEvent. So here is to cover `intersected()`: the rationale then is to make both ends a big number, i.e., M intersected with N, instead of M with 1, or 1 with N.

Reply:
try to put some of the rationale in the comment.

Reply:
done.